### ISSUE TITLE
Fix test_emitter.py on macOS

### ISSUE THREAD

**Issue by BoboTiG** [gorakhargosh/watchdog#546]
`test_emitter.py` is broken on macOS, we need to fix that.

**Comment by Ajordat** [comment]
The test `test_delete` also fails on Windows systems.

**Comment by evilham** [comment]
@BoboTiG could this have been accidentally fixed by #633?

**Comment by BoboTiG** [comment]
Nope, tests are simply skipped for now: https://github.com/gorakhargosh/watchdog/blob/9ac83bb71768b25cdf77ac6653a73adc91c916db/tests/test_emitter.py#L44-L45

**Comment by SamSchott** [comment]
The test failures don't look too bad with 0.10.3:

```
tests/test_emitter.py::test_create PASSED                                                                                                                                                                  [  3%]
tests/test_emitter.py::test_delete PASSED                                                                                                                                                                  [  4%]
tests/test_emitter.py::test_modify PASSED                                                                                                                                                                  [  6%]
tests/test_emitter.py::test_move PASSED                                                                                                                                                                    [  7%]
tests/test_emitter.py::test_move_to PASSED                                                                                                                                                                 [  8%]
tests/test_emitter.py::test_move_to_full SKIPPED                                                                                                                                                           [  9%]
tests/test_emitter.py::test_move_from PASSED                                                                                                                                                               [ 11%]
tests/test_emitter.py::test_move_from_full SKIPPED                                                                                                                                                         [ 12%]
tests/test_emitter.py::test_separate_consecutive_moves FAILED                                                                                                                                              [ 13%]
tests/test_emitter.py::test_delete_self FAILED                                                                                                                                                             [ 14%]
tests/test_emitter.py::test_fast_subdirectory_creation_deletion FAILED                                                                                                                                     [ 16%]
tests/test_emitter.py::test_passing_unicode_should_give_unicode PASSED                                                                                                                                     [ 17%]
tests/test_emitter.py::test_passing_bytes_should_give_bytes PASSED                                                                                                                                         [ 18%]
tests/test_emitter.py::test_recursive_on FAILED                                                                                                                                                            [ 19%]
tests/test_emitter.py::test_recursive_off FAILED                                                                                                                                                           [ 20%]
tests/test_emitter.py::test_renaming_top_level_directory FAILED                                                                                                                                            [ 22%]
tests/test_emitter.py::test_renaming_top_level_directory_on_windows SKIPPED                                                                                                                                [ 23%]
tests/test_emitter.py::test_move_nested_subdirectories FAILED                                                                                                                                              [ 24%]
tests/test_emitter.py::test_move_nested_subdirectories_on_windows SKIPPED
```

The failing tests mostly appear to come from the events being emitted in the wrong order. But this is not surprising when using an emitter that takes the difference of directory snapshots to generate events.

**Comment by BoboTiG** [comment]
@SamSchott I would love to merge a PR to enable and fix those tests :)

**Comment by SamSchott** [comment]
Well, I suspect that many of them will be fixed automatically with the switch to FSEvents (if the emitter no longer crashes). So the first step will be fix of #702.

I'm happy to try and clean up any remaining test failures.

**Comment by BoboTiG** [comment]
We are lacking of time here, so if you can give a shot, I would be happy 👍 

Given the regression you are encountering, I will be able to do a new release ASAP when it will be fixed.

**Comment by BoboTiG** [comment]
Closed by #734.
